include/mbedtls/config.h

```diff
@@ -256,6 +256,25 @@
  */
 //#define MBEDTLS_DEPRECATED_REMOVED
 
+/**
+ * \def MBEDTLS_CHECK_PARAMS
+ *
+ * This configuration controls whether the library validates parameters passed
+ * to it.
+ *
+ * Application code that deals with 3rd party input may wish to enable such
+ * validation, whilst code on closed systems, such as embedded systems, where
+ * the input is controlled and predictable, may wish to disable it entirely to
+ * reduce the code size of the library.
+ *
+ * When the symbol is not defined, no parameter validation except that required
+ * to ensure the integrity or security of the library are performed.
+ *
+ * When the symbol is defined, all parameters will be validated, and an error
+ * code returned where appropriate.
```

Comment:
We must explain in more detail how the user is supposed to configure failure callbacks. To my understanding, things are meant as follows:
```
If this option is set, the user must either define the parameter validation failure macro #MBEDTLS_PARAM_FAILED 
documented below, or provide a function mbedtls_param_failed() documented in platform_util.h.
```

Reply:
For later: there are a number of changes I want to suggest in that documentation, but we can leave them for later.